plugins/source/alicloud/resources/services/bss/bill_details_fetch.go

```diff
@@ -0,0 +1,113 @@
+	}
+	for _, billingCycle := range billingCycles {
+		billingDates := getDetailsBillingDates(billingCycle)
+		for _, billingDate := range billingDates {
+			request := bssopenapi.CreateQueryInstanceBillRequest()
+			request.BillingCycle = billingCycle
+			pageNum := 1
+			total := 0
+			DetailsmaxLimit := 100
+			request.PageNum = requests.NewInteger(pageNum)
+			request.BillingDate = billingDate
+			request.PageSize = requests.NewInteger(DetailsmaxLimit)
+			request.Granularity = "DAILY"
+			for {
+				response, err := c.Services().BSS.QueryInstanceBill(request)
+				if err != nil {
+					return err
+				}
+				if !response.Success {
+					code := response.GetHttpStatus()
+					return fmt.Errorf("got response status code %d (%v)", code, http.StatusText(code))
+				}
+				for _, item := range response.Data.Items.Item {
+					res <- &BillDetailsModel{
+						BillingCycle:          response.Data.BillingCycle,
+						BillingDate:           item.BillingDate,
+						AccountID:             response.Data.AccountID,
+						AccountName:           response.Data.AccountName,
+						ProductName:           item.ProductName,
+						SubOrderId:            item.SubOrderId,
+						DeductedByCashCoupons: item.DeductedByCashCoupons,
+						PaymentTime:           item.PaymentTime,
+						PaymentAmount:         item.PaymentAmount,
+						DeductedByPrepaidCard: item.DeductedByPrepaidCard,
+						InvoiceDiscount:       item.InvoiceDiscount,
+						UsageEndTime:          item.UsageEndTime,
+						Item:                  item.Item,
+						SubscriptionType:      item.SubscriptionType,
+						PretaxGrossAmount:     item.PretaxGrossAmount,
+						Currency:              item.Currency,
+						CommodityCode:         item.CommodityCode,
+						UsageStartTime:        item.UsageStartTime,
+						AdjustAmount:          item.AdjustAmount,
+						Status:                item.Status,
+						DeductedByCoupons:     item.DeductedByCoupons,
+						RoundDownDiscount:     item.RoundDownDiscount,
+						ProductDetail:         item.ProductDetail,
+						ProductCode:           item.ProductCode,
+						ProductType:           item.ProductType,
+						OutstandingAmount:     item.OutstandingAmount,
+						PipCode:               item.PipCode,
+						PretaxAmount:          item.PretaxAmount,
+						OwnerID:               item.OwnerID,
+						RecordID:              item.RecordID,
+						ResourceGroup:         item.ResourceGroup,
+						InstanceID:            item.InstanceID,
+						CashAmount:            item.CashAmount,
+					}
+				}
+				total += len(response.Data.Items.Item)
+				if len(response.Data.Items.Item) == 0 || total >= response.Data.TotalCount {
+					break
+				}
+				pageNum++
+				request.PageNum = requests.NewInteger(pageNum)
+			}
+		}
+	}
+	return nil
+}
+
+func getDetailsBillingCycles() []string {
+	var months []string
+	curMonth := time.Now().Format("2006-01")
+	months = append(months, curMonth)
+	lastMonth := time.Now().AddDate(0, 0, -5).Format("2006-01")
```

Comment:
I think this should match what we have in `bill_fetch.go`, which is using `(0, 0, -5)`: https://github.com/cloudquery/cloudquery/blob/main/plugins/source/alicloud/resources/services/bss/bill_fetch.go#L93

Otherwise, bills and bill_details could return results for different time periods. I think the idea was to fetch the previous month for the first 5 days of a new month, when the user is not requesting historical data. It's fine if we want to change this to always fetch the previous month, but let's then also update `bill_fetch.go`. What do you think @easayliu ?